app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  current_power do # Put this here for CLI
+    Power.new(current_user)
+  end
+
+  power :viewable_samples, as: :samples_scope
+
+  # first we grab the Sample from the id parameter passed to us
+  before_action :set_sample
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def index
+    @pipeline_run = select_pipeline_run(@sample, params[:pipeline_version])
+    @amr_counts = nil
+    can_see_amr = (current_user.admin? || current_user.allowed_feature_list.include?("AMR"))
```

Comment:
You probably don't need this right now, since you already have `before_action : admin_required` above. You can always add the feature flag when it's time to remove `admin_required`.

As a side note, it's better to call your feature flag "amr_heatmap", instead of re-using "AMR" from the old amr table, since we'll want to control access to the two features separately.

Reply:
My 2 cents: I prefer feature flags to admin (which was our old default way to guard features). Feature flag is more granular and will allow us to test locally with both test and admin accounts, and to enable it for key users for external review and usability research.

Reply:
I removed the feature flag since Mark is right, we already have `before_action :admin_required` restricting access and the feature flag can always be added back in later.